### DISCUSSION TITLE
Act uses github.event.inputs.variable_name but Github Actions uses inputs.variable_name?

### DISCUSSION THREAD

**Discussion by robertherber** [nektos/act#1303] [Q&A]
I just started using act. It's great to be able to test workflows locally.

But one things I just can't seem to get my head around is the inconsistency between act requiring `github.event.inputs` for inputs and Github Actions using just `inputs` for workflow_dispatch (tried with both workflow_dispatch and workflow_call on the GA side). What am I missing here? Is there a reason for this and is there a way around it?

Live example here. My current workaround is renaming all references to `inputs` to match Github Actions, but now I can obviously not run my workflows locally:
1. [Failing GA run because of using `github.event.inputs`](https://github.com/Kingstinct/utils/actions/runs/2855601678)
2. [Commit aligning workflows with Github Actions, act is now broken](https://github.com/Kingstinct/utils/commit/83d28cfe125af0fa218f4f9683aa8325f170a964)
3. [Successful GA run after commit](https://github.com/Kingstinct/utils/actions/runs/2855606655)

**Comment by KnisterPeter** [comment]
The event `workflow_dispatch` is not fully supported.
Using `github.event...` should work with GitHub as well, therefore it's not inconsistent but incomplete.
We are up for contributions to fill the gap.

**Comment by robertherber** [comment]
Well, `github.event.inputs` doesn't seems to work with Github (as you can see in the linked example). From what I see only `inputs` are [mentioned in the docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#providing-inputs).

If workflow_dispatch is not supported - what's the recommended way to test workflows..? I thought that'd be the most natural way to test one-off workflows.

**Reply by KnisterPeter** [reply]
As I've told, it's not fully supported. Obviously since you spotted a bug or incompatibility.
If you need it, please send a PR, which is more than welcome. 😊
Any help with act is greatly appreciated.

**Reply by robertherber** [reply]
> As I've told, it's not fully supported. Obviously since you spotted a bug or incompatibility.
> If you need it, please send a PR, which is more than welcome. 😊
> Any help with act is greatly appreciated.

Gotcha 👍🙂 My real use case here is to test reusable workflows, which made workflow_dispatch feel like a natural way to test it (since it similarly with inputs etc). Is there another way/trigger that you would recommend rather than using workflow_dispatch?

**Reply by KnisterPeter** [reply]
And exactly reusable workflow are not supported yet. Therefore everything would only be a workaround.
Haven't yet used them myself, so I cannot recommend a good testing strategy. I'm sorry
